Mathlib/Data/Matrix/Composition.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2024 Yunzhou Xie. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Yunzhou Xie
+-/
+import Mathlib.Data.Matrix.Basic
+
+/-!
+# Composition of matrices
+This file shows that Mₙ(Mₘ(R)) ≃ Mₙₘ(R), Mₙ(Mₘ(R)) ≃ Mₘ(Mₙ(R)), Mn(Rᵒᵖ) ≃ₐ[K] Mₙ(R)ᵒᵖ
+and also different levels of equivalence when R is an AddCommMonoid,
+Semiring, and Algebra over a CommSemiring K.
+-/
+namespace Matrix
+
+open BigOperators
+
+variable  (I J K L R : Type*)
+
+/-- Mₙ(Mₘ(R)) ≃ Mₙₘ(R) -/
+def comp : Matrix I J (Matrix K L R) ≃ Matrix (I × K) (J × L) R where
```

Comment:
Should we have some `simp` lemma such as `comp_apply` expressing what the ij-th entry look like

Reply:
that would be way over 150 lines

Reply:
Maybe you can tag the def as @[simps] to generate it automatically

Reply:
good point! done